near-sdk/src/environment/env.rs

```diff
@@ -279,6 +279,37 @@ pub fn keccak512_hash(value: &[u8]) -> [u8; 64] {
     }
 }
 
+/// Recovers an ECDSA signer address from a 32-byte message `hash` and a corresponding `signature`
+/// along with `v` recovery byte.
+///
+/// Takes in an additional flag to check for malleability of the signature
+/// which is generally only ideal for transactions.
+///
+/// Returns 64 bytes representing the public key if the recovery was successful.
+pub fn ecrecover(
```

Comment:
just one last thing about this, I think it's worth adding the unstable feature flag, just because we might want to change this to have a different interface in the future, and it's unclear if this is helpful to devs as is
```suggestion
#[cfg(feature = "unstable")]
pub fn ecrecover(
```

If you make this change, remember to add the feature flag to the test using this